Apps/W1/EDocumentConnectors/Tietoevry/app/src/Events.Codeunit.al

```diff
@@ -0,0 +1,115 @@
+                Error(this.MissingCustInfoErr, Customer.FieldCaption("VAT Registration No."), Customer."No.");
+
+        if not ServiceParticipant.Get(EDocumentService.Code, ServiceParticipant."Participant Type"::Customer, SalesHeader."Bill-to Customer No.") then
+            ServiceParticipant.Init();
+
+        if ServiceParticipant."Participant Identifier" = '' then
+            Error(this.MissingCustInfoErr, ServiceParticipant.FieldCaption("Participant Identifier"), Customer."No.");
+    end;
+
+    [EventSubscriber(ObjectType::Table, Database::"E-Document Service", 'OnAfterValidateEvent', 'Document Format', false, false)]
+    local procedure OnAfterValidateDocumentFormat(var Rec: Record "E-Document Service"; var xRec: Record "E-Document Service"; CurrFieldNo: Integer)
+    var
+        EDocServiceSupportedType: Record "E-Doc. Service Supported Type";
+    begin
+        if Rec."Document Format" <> Rec."Document Format"::"Tietoevry PEPPOL BIS 3.0" then
+            exit;
+
+        EDocServiceSupportedType.SetRange("E-Document Service Code", Rec.Code);
+        if EDocServiceSupportedType.IsEmpty() then begin
+            EDocServiceSupportedType.Init();
+            EDocServiceSupportedType."E-Document Service Code" := Rec.Code;
+            EDocServiceSupportedType."Source Document Type" := EDocServiceSupportedType."Source Document Type"::"Sales Invoice";
+            EDocServiceSupportedType.Insert();
+
+            EDocServiceSupportedType."Source Document Type" := EDocServiceSupportedType."Source Document Type"::"Sales Credit Memo";
+            EDocServiceSupportedType.Insert();
+
+            EDocServiceSupportedType."Source Document Type" := EDocServiceSupportedType."Source Document Type"::"Service Invoice";
+            EDocServiceSupportedType.Insert();
+
+            EDocServiceSupportedType."Source Document Type" := EDocServiceSupportedType."Source Document Type"::"Service Credit Memo";
+            EDocServiceSupportedType.Insert();
+        end;
+    end;
+
+    [EventSubscriber(ObjectType::Table, Database::"E-Document Service", 'OnAfterValidateEvent', "Service Integration V2", false, false)]
+    local procedure OnAfterValidateServiceIntegrationV2(var Rec: Record "E-Document Service"; var xRec: Record "E-Document Service"; CurrFieldNo: Integer)
+    begin
+        if Rec."Service Integration V2" <> Rec."Service Integration V2"::Tietoevry then
+            exit;
+        Rec.Validate("Document Format", Rec."Document Format"::"Tietoevry PEPPOL BIS 3.0");
+        Rec.Modify(true);
+    end;
+
+    [EventSubscriber(ObjectType::Table, Database::"Service Participant", 'OnAfterValidateEvent', 'Participant Identifier', false, false)]
+    local procedure OnAfterValidateServiceParticipant(var Rec: Record "Service Participant"; var xRec: Record "Service Participant"; CurrFieldNo: Integer)
+    var
+        EDocumentService: Record "E-Document Service";
+    begin
+        if not EDocumentService.Get(Rec.Service) then
```

Comment:
Dont you want to fail here?

Reply:
I don't want to execute the code if tietoevry service is not setup anyway..